File: cogs/utility/name_history_cog.py
import discord
from discord.ext import commands
from utils.database import fetch_name_history
from utils.error_handler import error_handler
from utils.permissions import has_permissions_to_send_messages

class NameHistoryCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self.local_cache = bot.local_cache

    @commands.command(name='namehistory')
    @error_handler
    @has_permissions_to_send_messages()
    async def name_history(self, ctx, user: discord.User = None):
        if user is None:
            user = ctx.author
        name_history = self.fetch_name_history(user.id)
        if name_history:
            await ctx.send(f"Here's the name history for {user.name}: {', '.join(name_history)}")
        else:
            await ctx.send(f"Sorry, I couldn't find any name history for {user.name}.")

    def fetch_name_history(self, user_id):
        return fetch_name_history(user_id, self.local_cache)

def setup(bot):
    bot.add_cog(NameHistoryCog(bot))

File: cogs/utility/name_history_cog.ts
import { Client, Message, User } from 'discord.js-selfbot-v13';
import { Command } from 'discord-akairo';
import { fetchNameHistory } from '../../utils/database';
import { errorHandler } from '../../utils/error_handler';
import { hasPermissionsToSendMessages } from '../../utils/permissions';

class NameHistoryCog {
    private client: Client;
    private localCache: any;

    constructor(client: Client) {
        this.client = client;
        this.localCache = client.localCache;
    }

    @Command({
        aliases: ['namehistory'],
        description: 'Fetch name history of a user.',
        args: [
            { id: 'user', type: 'user', default: null }
        ]
    })
    @errorHandler
    @hasPermissionsToSendMessages()
    async nameHistory(message: Message, { user }: { user: User | null }) {
        const targetUser = user || message.author;

        try {
            const nameHistory = this.fetchNameHistory(targetUser.id);
            if (nameHistory.length > 0) {
                await message.channel.send(`Here's the name history for ${targetUser.username}: ${nameHistory.join(', ')}`);
            } else {
                await message.channel.send(`Sorry, I couldn't find any name history for ${targetUser.username}.`);
            }
        } catch (error) {
            console.error(`Failed to fetch name history. Error: ${error.message}`);
            await message.channel.send(`An error occurred while fetching name history for ${targetUser.username}. Please try again later.`);
        }
    }

    fetchNameHistory(userId: string) {
        return fetchNameHistory(userId, this.localCache);
    }
}

export function setup(client: Client) {
    return new NameHistoryCog(client);
}
